rfcs/pagination-component.md

```diff
@@ -0,0 +1,70 @@
+# Pagination Component
+
+### Exports
+
+- `Pagination`
+- `StatefulPagination`
+- `StatefulContainer`
+- `StyledRoot`
+- `StyledBackButton`
+- `StyledNextButton`
+- `StyledMaxLabel`
+- `STATE_CHANGE_TYPE`
+
+### Pagination API
+
+- `numPages: number` - Required
+  Max number of pages
+- `currentPage: number` - Required
+  The current page
+- `prepositionLabel: string` - Optional
+  Defaults to "of"
```

Comment:
One thing we haven't thought a ton about is internationalization. In theory with `overrides` you can change everything, but it seems like having props like the one you have here would be very useful for that. Maybe we should extend this to allow customizing labels for "Prev" and "Next"?

[Atlaskit has an i18n object for this](https://atlaskit.atlassian.com/packages/core/pagination). It seems like a `labels: {next, prev, preposition}` object might be more generic. Or we could stay with `prepositionLabel` and add `prevLabel`, `nextLabel`. Thoughts?

Reply:
I was thinking that the user could just override the button itself directly (and pass in the labels that way) vs. surfacing a high level prop here. What do you think? If that's too much, then the object approach here is good.

Reply:
```js
<Pagination overrides={{
  StyledBackButton: props => (
    <StyledBackButton {...props}>{app.strings.previous}</Button>
  ),
  StyledNextButton: props => (
    <StyledNextButton {...props}>{app.strings.next}</Button>
  )
}}>
```

That's not too bad.

I looked a bit at other designs to see if other components will run into this. It seems like datepicker is one of the only other components that has _visible_ static labels (there's probably plenty of invisible aria labels for close/next buttons, etc).

![image](https://user-images.githubusercontent.com/875591/45051658-8b360d80-b039-11e8-9504-6350bacad6cd.png)

Overriding 7 components for the "Quick Select" and each button might get a bit tedious. The object format is a slight improvement on this.

All that said, I did a bit more research on i18n in component libraries, and I think [ant's approach](https://ant.design/docs/react/i18n) makes sense. Instead of passing in override strings at the callsite, you have a json file full of strings that are passed through via context. So maybe supporting overrides for all libraries here isn't so necessary. Feel free to just make a decision and land. I'll open up a separate issue about general i18n strategy.